streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -113,9 +121,19 @@ private void putAndMaybeForward(final ThreadCache.DirtyEntry entry,
         }
     }
 
+    @Override
+    public <FK, FV> FlushEntry<FK, FV> flushEntry(final StateSerdes<FK, FV> serdes,
```

Comment:
I agree that we should clean this up. But this should be done in a follow up PR: In fact, all caching stores should be non-generic `<Bytes,byte[]>` stores and we should not pass any `Serdes` to them to begin with. I would like to exclude this cleanup from this PR though.

If you want, I can also do a cleanup PR first before we merge this and rebase base on the cleanup PR. To avoid messy code in `trunk` as intermediate. Thoughts?

Reply:
Opened a cleanup PR: https://github.com/apache/kafka/pull/6331